### PR TITLE
Add `default_aggregate_function` parameter for AggregatingMergeTree

### PR DESCRIPTION
**Pull request by zvonand**
Add `default_aggregate_function` parameter for AggregatingMergeTree allowing to specify aggregate function for columns that have no explicit aggregation. A partial (syntax-only) solution for #57240

### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add `default_aggregate_function` parameter for AggregatingMergeTree allowing to specify aggregate function for columns that have no explicit aggregation.

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by zvonand** [comment]
@alexey-milovidov could anyone please put a label / take a look?

File: docs/en/engines/table-engines/mergetree-family/aggregatingmergetree.md [hunk 1]
```diff
@@ -35,6 +35,16 @@ CREATE TABLE [IF NOT EXISTS] [db.]table_name [ON CLUSTER cluster]
...
 For a description of request parameters, see [request description](../../../sql-reference/statements/create/table.md).
 
+### Parameters of AggregatingMergeTree
+
+#### default_aggregate_function
+
+`default_aggregate_function` - a name of a [SimpleAggregateFunction](../../../sql-reference/data-types/simpleaggregatefunction.md) to be applied to all columns whose type is different from [AggregateFunction](../../../sql-reference/data-types/aggregatefunction.md) or [SimpleAggregateFunction](../../../sql-reference/data-types/simpleaggregatefunction.md). Optional parameter.
+The function is not applied to columns that are in primary key.
+
+If `default_aggregate_function` is not specified, no aggregate function is applied by default.
```
Comment (UnamedRus) [inline docs/en/engines/table-engines/mergetree-family/aggregatingmergetree.md:45]: It's actually, does mean that `any` aggregate function is applied

**Comment by alexey-milovidov** [comment]
It looks too ad-hoc, the feature needs more thought...

**Comment by alexey-milovidov** [comment]
Keep in mind that I still don't like the existence of SimpleAggregateFunction.

**Comment by zvonand** [comment]
> Keep in mind that I still don't like the existence of SimpleAggregateFunction.

Why?  
But it does not really matter here. I can remove the limitation and then any aggregate function can be used. But IMO it is not a good thing that type of values stored in a column will be different from the explicitly declared ones. With SimpleAggregate, state is a value of the type.

**Comment by zvonand** [comment]
> It looks too ad-hoc, the feature needs more thought...

WDYM? Add more flexibility? Or make something different instead?  

One improvement I can think of is ability do define more than one default functions (in a tuple), and each of them will be applied to corresponding column.

File: docs/en/engines/table-engines/mergetree-family/aggregatingmergetree.md (hunk 1 above)
Comment (zvonand) [reply docs/en/engines/table-engines/mergetree-family/aggregatingmergetree.md]: Updated

**Comment by alexey-milovidov** [comment]
@zvonand A test for LDAP has failed: https://s3.amazonaws.com/clickhouse-test-reports/65562/57750540c4cbad4b3ea6719c14eb4611f67663f8/integration_tests__aarch64__[4_6].html

**Comment by zvonand** [comment]
> @zvonand A test for LDAP has failed: https://s3.amazonaws.com/clickhouse-test-reports/65562/57750540c4cbad4b3ea6719c14eb4611f67663f8/integration_tests__aarch64__[4_6].html
> 

But this is another PR and unrelated test  the CI is not even running here

**Comment by alexey-milovidov** [comment]
Maybe you can help fix it nevertheless, or find @traceon and make him fix it?

**Comment by pamarcos** [comment]
@zvonand the test for LDAP is fixed through https://github.com/ClickHouse/ClickHouse/pull/70755
Please merge master and fix conflicts to give the CI another try

**Comment by zvonand** [comment]
> Please merge master and fix conflicts to give the CI another try

@pamarcos there is no CI enabled here yet

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit d1e70fd9173347ff9d8a6c3050d7613cdf5d2d8d with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62707/d1e70fd9173347ff9d8a6c3050d7613cdf5d2d8d/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/62707/d1e70fd9173347ff9d8a6c3050d7613cdf5d2d8d/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by zvonand** [comment]
Hm, CI is very green now

**Comment by martin31821** [comment]
@zvonand Question - I'd like to use this to aggregate several smaller tables into a wider table with partial materialized views. In that case, I'd use anyLast() as aggregation function, where the documentation states that the ordering is undefined here. 

- Is anyLast() guaranteed in this scenario to overwrite old value on an incoming block with always the latest? 
- If not, is there any order by clause that can enforce this?

**Comment by zvonand** [comment]
@martin31821 this is a tricky thing. The correct order is preserved inside _one_ data portion, e.g. when you do:
```sql
INSERT INTO my_table VALUES (....), (....), (....);
```
then the order will be preserved.

In other cases the thing is that the order of merges in CH is not determined. Merge of block with newer data can happen before merging with an older block, and then the order would be different: older data will be considered the latest when in comes to `anyLast` itself. AFAIK there is no option to pre-determine merges sequence

**Comment by martin31821** [comment]
is there any way at all to achieve such a behavior right now? I think #57240 describes a different problem than what the default_aggregate_function is achieving. 

I think the issue calls for a different (possibly new) aggregation function such as `maxBy()` which allows to pick a single value based on a different column (think timestamp, version number), so we can reliably replace/aggregate a part of a row based on a later timestamp, which, as far as I've understood was what the issue was about.

**Pull request closed without merging**